backend/src/domain/mappers/DisciplineMapper.ts

```diff
@@ -0,0 +1,12 @@
+import { ITypeAndWorkloadDto } from '../interfaces/contracts/ITypeAndWorkloadDto'
+import { IDiscipline } from '../interfaces/entities/IDiscipline'
+import { ITypeDiscipline } from '../interfaces/entities/ITypeDicipline'
+
+export function toTypeAndWorkload (data: IDiscipline, typeDiscipline: ITypeDiscipline | undefined): ITypeAndWorkloadDto {
```

Comment:
Esse nome e o nome da Dto estão estranhos, não estão representando bem.
Tipo, se você for ler fica "type do que?"